### DISCUSSION TITLE
How to add sql hints to the generated query?

### DISCUSSION THREAD

**Discussion by krovicko** [mybatis/mybatis-dynamic-sql#867] [Q&A]
I need to add option (recompile) to my query, but the only possibility I found is to manually add it to the created statement (wrap the StatementProvider and change the getSelectStatement method). Would anyone have a better solution, some annotation in the mapper or something like that? I didn't find anything in the documentation... Thanks!

**Comment by harawata** [comment]
Hello @krovicko ,

- Which DB are you using?
- Please show us example SQL, with and without the option. We need to understand what you want to achieve.
- There is no `StatementProvider` in MyBatis, so I could not follow your explanation ("wrap the StatementProvider and change the getSelectStatement method"). Show us some code (as text, not picture).

**Reply by jeffgbutler** [reply]
I think this is a question for MyBatis Dynamic SQL. There's an open issue for this request: https://github.com/mybatis/mybatis-dynamic-sql/issues/453. I hope to add something to support this in version 2.x of the library.

**Reply by krovicko** [reply]
I am using MSSQL database, the sql command is easy, like that:

SELECT column1, column2, ...
FROM table_name
WHERE condition OPTION(RECOMPILE);

And it is really a question for MyBatis Dynamic SQL. Sorry. I was talking about org.mybatis.dynamic.sql.select.render.SelectStatementProvider.
